### PR TITLE
Action Manager | Migrate Entity Outliner and Viewport Context Menus

### PR DESCRIPTION
**Pull request by AMZN-daimini**
## What does this PR do?

Migrates the Entity Outliner and Viewport context menus to use the new Action Manager.
This should introduce no change in behavior, although it may result in changes in labeling for some actions due to them being re-used from other parts of the Editor while they previously weren't, conforming their naming.

Exception: this change temporarily removes the "Create multiplayer entity" menu item. It will be restored in a follow-up PR (as it requires some extra changes and this PR is big enough as it is).

## How was this PR tested?

Verified the behavior manually.

### No selection
*Outliner*
before_outliner_noselectionafter_outliner_noselection

*Viewport*
before_viewport_noselectionafter_viewport_noselection

### Entity selection
*Outliner*
before_outliner_entityselectionafter_outliner_entityselection

*Viewport*
before_viewport_entityselectionafter_viewport_entityselection

### Prefab selection
*Outliner*
before_outliner_prefabselectionafter_outliner_prefabselection

*Viewport*
before_viewport_prefabselectionafter_viewport_prefabselection

### Procedural Prefab selection
*Outliner*
before_outliner_procprefabselectionafter_outliner_procprefabselection

*Viewport*
before_viewport_procprefabselectionafter_viewport_procprefabselection

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Review by forkercat — APPROVED** [review]
nice!

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp [hunk 1]
```diff
@@ -766,6 +1358,143 @@ namespace AzToolsFramework
             );
         }
 
+        bool PrefabIntegrationManager::CanCreatePrefabWithCurrentSelection(const AzToolsFramework::EntityIdList& selectedEntities)
+        {
+            if(selectedEntities.empty())
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:1363]: uber nit: space after ` (`

**Review by hultonha — COMMENTED** [review]
Overall looks good, I have mostly minor feedback. Changes and consistency look great! 😄 (also can we update `Open pinned Inspector` to `Open Pinned Inspector` 😁 Thanks!

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp [hunk 2]
```diff
@@ -1213,6 +1228,26 @@ namespace AzToolsFramework
         EnableUi(enable);
     }
 
+    void EntityOutlinerWidget::RenameEntity(const AZ::EntityId& entityId)
```
[the remark this answers is not in the document]
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp:1231]: It triggers the rename user workflow on the UI side. I'll look for a better name!

File: Code/Editor/Core/EditorActionsHandler.cpp [hunk 3]
```diff
@@ -771,6 +775,107 @@ void EditorActionsHandler::OnActionRegistrationHook()
...
+        m_actionManagerInterface->InstallEnabledStateCallback(
+            actionIdentifier,
+            []() -> bool
+            {
+                AzToolsFramework::EntityIdList selectedEntities;
+                AzToolsFramework::ToolsApplicationRequests::Bus::BroadcastResult(
+                    selectedEntities, &AzToolsFramework::ToolsApplicationRequests::Bus::Events::GetSelectedEntities);
+
+                // Can only rename one entity at a time
+                if (selectedEntities.size() != 1)
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Editor/Core/EditorActionsHandler.cpp]: Awesome thanks! 😄

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/ContextMenuHandlers.cpp [hunk 4]
```diff
@@ -54,3 +61,71 @@ void ContextMenuBottomHandler::PopulateEditorGlobalContextMenu(
...
+
+        actionManagerInterface->InstallEnabledStateCallback(
+            actionIdentifier,
+            []() -> bool
+            {
+                AzToolsFramework::EntityIdList selectedEntities;
+                AzToolsFramework::ToolsApplicationRequests::Bus::BroadcastResult(
+                    selectedEntities, &AzToolsFramework::ToolsApplicationRequests::Bus::Events::GetSelectedEntities);
+
+                return selectedEntities.size() > 0;
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/ContextMenuHandlers.cpp]: Nice! 😁

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp [hunk 5]
```diff
@@ -1427,6 +1437,114 @@ void SandboxIntegrationManager::OnLayerComponentDeactivated(AZ::EntityId entityI
...
+            actionIdentifier,
+            actionProperties,
+            [this]()
+            {
+                AzToolsFramework::EntityIdList selectedEntities;
+                AzToolsFramework::ToolsApplicationRequests::Bus::BroadcastResult(
+                    selectedEntities, &AzToolsFramework::ToolsApplicationRequests::Bus::Events::GetSelectedEntities);
+
+                // when nothing is selected, entity is created at root level.
+                if (selectedEntities.size() == 0)
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp]: Thanks!

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp [hunk 6]
```diff
@@ -1427,6 +1437,114 @@ void SandboxIntegrationManager::OnLayerComponentDeactivated(AZ::EntityId entityI
...
+
+}
+
+void SandboxIntegrationManager::OnMenuBindingHook()
+{
+    auto menuManagerInterface = AZ::Interface<AzToolsFramework::MenuManagerInterface>::Get();
+
+    if (!menuManagerInterface)
+    {
+        // TODO - Assert?
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp]: Thanks! 🥳

File: Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntitySortComponent.cpp [hunk 7]
```diff
@@ -244,6 +244,82 @@ namespace AzToolsFramework
             return entityItr != m_childEntityOrderCache.end() ? entityItr->second : std::numeric_limits<AZ::u64>::max();
         }
 
+        bool EditorEntitySortComponent::CanMoveChildEntityUp(const AZ::EntityId& entityId)
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntitySortComponent.cpp:247]: Nice one thanks!

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp]: Yes understood, I just think a new name would help, thanks!

**Pull request merged**